.github/workflows/pre-commit.yml

```diff
@@ -0,0 +1,17 @@
+name: code-format-check
+
+on:
+  push:
+    branches: [ "master" ]
+  pull_request:
+    branches: [ "master" ]
+
+jobs:
+  pre-commit:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+    - uses: actions/setup-python@v5
+      with:
+        python-version: '3.11'
```

Comment:
The python 3.11 is not a mandate as of now.

Reply:
Because we use py3.11 for CICD environment，so we use the same version for pre-commit